CorsixTH/Lua/dialogs/menu.lua

```diff
@@ -677,17 +677,10 @@ function UIMenuBar:makeMapeditorMenu(app)
   local menu = UIMenu()
   local hotkeys = app.hotkeys
 
-  local function mapeditorSave(ui)
-    local message = app.world:validateMap()
-    if message then
-      ui:addWindow(UIInformation(ui, {message}))
-    end
-    ui:addWindow(UISaveMap(ui))
-  end
   menu:appendItem(_S.menu_file.load:format(hotkey_value_label("ingame_loadMenu", hotkeys)), function() self.ui:addWindow(UILoadMap(self.ui, "map")) end)
-    :appendItem(_S.menu_file.save:format(hotkey_value_label("ingame_saveMenu", hotkeys)), function() mapeditorSave(self.ui) end)
-    :appendItem(_S.menu_file.restart:format(hotkey_value_label("ingame_restartLevel", hotkeys)), function() app:mapEdit() end)
-    :appendItem(_S.menu_file.quit:format(hotkey_value_label("ingame_quitLevel", hotkeys)), function() self.ui:quit() end)
+    :appendItem(_S.menu_file.save:format(hotkey_value_label("ingame_saveMenu", hotkeys)), function() self.ui:saveMapeditor() end)
```

Comment:
Shouldn't validation be able to be handled by the UISaveMap class? Or more, should it belong there instead?

Reply:
Done that, though it may break a design rule about logic in a UI class.

Reply:
@TheCycoONE can you advise? There's not a clear-cut answer I think as to where the call and code should exist.

Reply:
Would the `Map` class  work?

It seems to mostly handle the loaded map, although there is also other stuff in there such as loading level config, it seems.

Alternatively, something `LoadSave`, although it may be too early for that.